pkg/provider/aws/resources/eks_test.go

```diff
@@ -0,0 +1,103 @@
+package resources
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_CreateEksCluster(t *testing.T) {
+	appName := "test-app"
+	clusterName := "test-cluster"
+	eus := []*core.ExecutionUnit{{AnnotationKey: core.AnnotationKey{ID: "test"}}}
+	subnet := NewSubnet("test-subnet", NewVpc(appName), "", PrivateSubnet)
+	type stringDep struct {
+		source string
+		dest   string
+	}
+	type testResult struct {
+		nodes []string
+		deps  []stringDep
+	}
+	cases := []struct {
+		name string
+		want testResult
+	}{
+		{
+			name: "happy path",
+			want: testResult{
+				nodes: []string{"aws:eks_cluster:test-app-test-cluster",
+					"aws:eks_fargate_profile:test-app-test-cluster",
+					"aws:eks_node_group:test-app-test-cluster",
+					"aws:iam_role:test-app-test-cluster-FargateExecutionRole",
+					"aws:iam_role:test-app-test-cluster-NodeGroupRole",
+					"aws:iam_role:test-app-test-cluster-k8sAdmin",
+				},
+				deps: []stringDep{
+					{dest: "aws:eks_cluster:test-app-test-cluster", source: "aws:eks_fargate_profile:test-app-test-cluster"},
+					{dest: "aws:eks_cluster:test-app-test-cluster", source: "aws:eks_node_group:test-app-test-cluster"},
+					{dest: "aws:iam_role:test-app-test-cluster-k8sAdmin", source: "aws:eks_cluster:test-app-test-cluster"},
+					{dest: "aws:iam_role:test-app-test-cluster-FargateExecutionRole", source: "aws:eks_fargate_profile:test-app-test-cluster"},
+					{dest: "aws:iam_role:test-app-test-cluster-NodeGroupRole", source: "aws:eks_node_group:test-app-test-cluster"},
+				},
+			},
+		},
+	}
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			assert := assert.New(t)
+			dag := core.NewResourceGraph()
+			CreateEksCluster(appName, clusterName, []*Subnet{subnet}, nil, eus, dag)
+			for _, id := range tt.want.nodes {
+				resource := dag.GetResource(id)
+				assert.NotNil(resource, fmt.Sprintf("Resource %s, not found", id))
+				assert.Equal(resource.KlothoConstructRef(), []core.AnnotationKey{{ID: "test"}}, fmt.Sprintf("Resource %s, did not recieve klotho construct ref", resource.Id()))
```

Comment:
minor: I don't think you need the `fmt.Sprintf` — you can just write the format string and `resource.Id()` directly. (You _may_ need to turn this into `Equalf` for that to work, though as far as I can tell they seem to be effectively the same?)

Reply:
cool didnt know equalF was a thing